include/mbedtls/sm3.h

```diff
@@ -0,0 +1,177 @@
+/**
+ * \file sm3.h
+ *
+ * \brief SM3 message digest algorithm (hash function)
+ */
```

Comment:
Please add a reference to the specification of SM3 (like there is at the top of `sha256.h` and so on).